### ISSUE TITLE
[JENKINS-18778] Builds disappear from build history

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#14744]
<p>We have 4-5 projects (out of ~40 projects) that all the builds disappear from the "Build History" from time to time.  There was no external event (like restarting Jenkins, or triggering a new build for those projects).<br/>
We know those builds are there from the Jobs directory, but they just don't show on the Build history.</p>

<p>Only after we restart Jenkins, and after restart the builds come to the build history.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ctlam">ctlam</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778" target="_blank">Builds disappear from build history</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2014-05-18T12:44:41+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We have 4-5 projects (out of ~40 projects) that all the builds disappear from the "Build History" from time to time.  There was no external event (like restarting Jenkins, or triggering a new build for those projects).
We know those builds are there from the Jobs directory, but they just don't show on the Build history.</p>

<p>Only after we restart Jenkins, and after restart the builds come to the build history.</p>
</pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins 1.521</code></li></ul>
<details><summary><i>3 attachments</i></summary>

- [screenshot-1.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24004/screenshot-1.jpg)
  > ![screenshot-1.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24004/screenshot-1.jpg)
- [screenshot-2.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24005/screenshot-2.jpg)
  > ![screenshot-2.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24005/screenshot-2.jpg)
- [screenshot-3.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24006/screenshot-3.jpg)
  > ![screenshot-3.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24006/screenshot-3.jpg)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>prsgan:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778?focusedId=181789&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181789">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The similar issue also noticed with version 1.509.2</p>
</pre>
</details>

</ul>
</details>

<p>The similar issue also noticed with version 1.509.2</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ctlam:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778?focusedId=182777&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-182777">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>New information:
I can reproduce it more consistently now:
After you rename your project, the project will lose out the build history.
Restarting Jenkins will resolve the problem.</p>
</pre>
</details>

</ul>
</details>

<p>New information:<br/>
I can reproduce it more consistently now:<br/>
After you rename your project, the project will lose out the build history.<br/>
Restarting Jenkins will resolve the problem.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/psftw.svg" title="psftw's avatar" /> psftw:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778?focusedId=186232&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-186232">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I can confirm on 1.531 that this issue only seems to happen when I've renamed a job and have not restarted Jenkins yet.  After a restart, the job history returns.</p>
</pre>
</details>

</ul>
</details>

<p>I can confirm on 1.531 that this issue only seems to happen when I've renamed a job and have not restarted Jenkins yet.  After a restart, the job history returns.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/psftw.svg" title="psftw's avatar" /> psftw:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778?focusedId=186240&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-186240">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Downgrading to minor due to easy workaround and no loss of data.</p>
</pre>
</details>

</ul>
</details>

<p>Downgrading to minor due to easy workaround and no loss of data.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778?focusedId=200569&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200569">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>What about <a href="https://issues.jenkins.io/browse/JENKINS-18678" title="Builds disappear some time after renaming job" class="issue-link" data-issue-key="JENKINS-18678"><del>JENKINS-18678</del></a>?
Do you see the issue on newer versions?</p>
</pre>
</details>

</ul>
</details>

<p>What about <a href="https://issues.jenkins.io/browse/JENKINS-18678" title="Builds disappear some time after renaming job" class="issue-link" data-issue-key="JENKINS-18678"><del>JENKINS-18678</del></a>?<br/>
Do you see the issue on newer versions?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-18778?focusedId=200633&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200633">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Based on comment by requester, this is clearly a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-18678" title="Builds disappear some time after renaming job" class="issue-link" data-issue-key="JENKINS-18678"><del>JENKINS-18678</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>Based on comment by requester, this is clearly a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-18678" title="Builds disappear some time after renaming job" class="issue-link" data-issue-key="JENKINS-18678"><del>JENKINS-18678</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a href="https://github.com/jenkinsci/jenkins/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-18678%22">JENKINS-18678</a>]</i>
